comid/rawint.go

```diff
@@ -0,0 +1,330 @@
+		return ""
+	}
+
+	return o.Value.String()
+}
+
+// MarshalJSON serializes RawInt into JSON
+func (o RawInt) MarshalJSON() ([]byte, error) {
+	valueBytes, err := json.Marshal(o.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	value := encoding.TypeAndValue{Type: o.Value.Type(), Value: valueBytes}
+
+	return json.Marshal(value)
+}
+
+// UnmarshalJSON de-serializes input JSON data into RawInt
+func (o *RawInt) UnmarshalJSON(data []byte) error {
+	var tnv encoding.TypeAndValue
+
+	if err := json.Unmarshal(data, &tnv); err != nil {
+		return err
+	}
+
+	decoded, err := NewRawInt(nil, tnv.Type)
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(tnv.Value, decoded.Value); err != nil {
+		return err
+	}
+
+	if err := decoded.Value.Valid(); err != nil {
+		return fmt.Errorf("invalid %s: %w", tnv.Type, err)
+	}
+
+	o.Value = decoded.Value
+
+	return nil
+}
+
+// MarshalCBOR serializes RawInt to CBOR
+func (o RawInt) MarshalCBOR() ([]byte, error) {
+	return em.Marshal(o.Value)
+}
+
+// UnmarshalCBOR de-serializes input CBOR into RawInt
+func (o *RawInt) UnmarshalCBOR(data []byte) error {
+	if len(data) == 0 {
+		return errors.New("empty input")
+	}
+
+	majorType := (data[0] & 0xe0) >> 5
+	switch majorType {
+	case 0, 1:
+		rawIntInteger := new(RawIntInteger)
+		if err := dm.Unmarshal(data, rawIntInteger); err != nil {
+			return err
+		}
+		o.Value = rawIntInteger
+	case 6:
+		rawIntRange := new(TaggedRawIntRange)
+		if err := dm.Unmarshal(data, rawIntRange); err != nil {
+			return err
+		}
+		o.Value = rawIntRange
+	default:
+		return fmt.Errorf("RawInt: Error: unknown major type: %d", majorType)
+	}
+
+	return nil
+}
+
+// RawIntIntegerType is the name of the Integer version of RawInt
+const RawIntIntegerType = "rawIntInteger"
+
+// RawIntInteger implements the Integer version of RawInt. It can
+// hold both positive and negative integer values
+type RawIntInteger int64
+
+// NewRawIntInteger creates a RawIntInteger from the given input value
+func NewRawIntInteger(val any) (*RawIntInteger, error) {
+	var ret RawIntInteger
+
+	if val == nil {
+		return &ret, nil
+	}
+
+	switch v := val.(type) {
+	case RawIntInteger:
+		ret = v
+	case *RawIntInteger:
+		ret = *v
+	case int64:
+		ret = RawIntInteger(v)
+	default:
+		return nil, fmt.Errorf("unexpected type for RawIntInteger: %T", v)
+	}
+
+	return &ret, nil
+}
+
+// Valid is a no-op for RawIntInteger and always returns nil
+func (o RawIntInteger) Valid() error { return nil }
+
+// String converts RawIntInteger into a string
+func (o RawIntInteger) String() string { return fmt.Sprint(int64(o)) }
+
+// Type returns the name/type of RawIntInteger
+func (o RawIntInteger) Type() string {
+	return RawIntIntegerType
+}
+
+// CompareAgainstRefInteger compares RawIntInteger against another RawIntInteger reference value
+func (o RawIntInteger) CompareAgainstRefInteger(ref RawIntInteger) bool {
+	return o == ref
+}
+
+// CompareAgainstRefRange compares RawIntInteger against a TaggedRawIntRange
```

Comment:
This should describe the comparison rule. An Int is equal to an IntRange iff the int range min and max are both the same as the int.